spec/requests/saved_pages_request_spec.rb

```diff
@@ -0,0 +1,45 @@
+RSpec.describe "SavedPages", type: :request do
+  context "when receiving an unauthenticated request" do
+    it "returns unauthorised for GET /api/saved_pages" do
+      get saved_pages_path
+
+      expect(response).to have_http_status(:unauthorized)
+    end
+
+  end
+
+  context "when receiving an authenticated request" do
+    before do
+      stub_oidc_discovery
+
+      # rubocop:disable RSpec/AnyInstance
+      allow_any_instance_of(OidcClient).to receive(:tokens!).and_return({ access_token: "access-token", refresh_token: "refresh-token" })
+      # rubocop:enable RSpec/AnyInstance
```

Comment:
An alternative to disabling `RSpec/AnyInstance` could be to use an instance double instead of stubbing any instance of `OidcClient`:
```ruby
# outside the before block
let(:oidc_client) { instance_double(OidcClient) }

# ...

# inside the before block
allow(OidcClient).to receive(:new).and_return(oidc_client)
allow(oidc_client).to receive(:tokens!).and_return({ access_token: "access-token", refresh_token: "refresh-token" })
```

This seems to be the approach [suggested by rubocop](https://www.rubydoc.info/gems/rubocop-rspec/1.6.0/RuboCop/Cop/RSpec/AnyInstance).

Reply:
great find! I'll amend these and also go back and amend the ones in account_api in a final commit